### DISCUSSION TITLE
$h͏_import is not defined

### DISCUSSION THREAD

**Discussion by tri2820** [endojs/endo#2638] [Q&A]
Hi all, I'm having `$h͏_import is not defined` error when importing module using dynamic import. This is the minimal snippet to reproduce the error:

```js
const compartment = new Compartment({
      __options__: true,
      importNowHook(importSpecifier: string, referrerSpecifier: string) {
        if (importSpecifier === "test") {
          // dynamic import
          const source = new ModuleSource(`import("dep");`);
          return {
            source,
            specifier: importSpecifier,
            compartment, // reflexive
          };
        }
      },
    });

    compartment.importNow("test");
```

I believe SES transforms dynamic import statements to `$h_import` statements. If so, how to provide the implementation of `$h_import` function?

The only related issue I found is https://github.com/endojs/endo/issues/291 - is the feature work-in-progress?

**Comment by kriskowal** [comment]
This is feature work in progress. I can have this implemented with tests in short order.

**Reply by kriskowal** [reply]
This finishes the implementation of dynamic import https://github.com/endojs/endo/pull/2639

You would be most welcome in our Matrix channel https://matrix.to/#/#endojs:matrix.org
